### ISSUE TITLE
Black screen when FS-UAE is started from FS-UAE Arcade

### ISSUE THREAD

**Issue by FrodeSolheim** [FrodeSolheim/fs-uae#76]
One user (Alzabo) has reported:
"I get a black screen with sound working if I run any game through FS-UAE Arcade but they work fine if I run them through FS-UAE Launcher…what could it be?

WIndows 10 Home
AMD E-450 APU with Radeon 6320 HD
Latest AMD Drivers"

**Comment by mak77** [comment]
hi, I can reproduce this bug, AMD A6 - 7400K, connected through hdmi, Catalyst 15.8 beta. is it maybe drawing to the wrong monitor?
No problems from the launcher.

**Comment by FrodeSolheim** [comment]
<CoX> @FrodeSolheim: i can reproduce this with all later versions of FS UAE arcade
<CoX> only works (flawlessly though :D) on fs-uae-suite_2.4.1+portable_windows
<CoX> GPU: Radeon HD6490M __  DRIVER INFO: from: 30-09-2011 Driver ver. 8.882.2.3000

**Comment by MatteAce** [comment]
I have exactly the same bug, both in 2.6 and 2.7 DEV. I had to rollback to 2.5 in order to have a usable Arcade, but then I've lost the updated WHDload (without the nag screen)

I've also noticed (I don't know if it's related) that I can WIN+D to show the desktop on 2.5 but if I try that on 2.7 the screen freezes and I need to kill the app with task manager.

I'm on Win10, AMD A8, Radeon R9 380

**Comment by FrodeSolheim** [comment]
Hi @MatteAce, just to be sure:
- Is FS-UAE Arcade itself black https://github.com/FrodeSolheim/fs-uae-arcade/issues/7
- Or it FS-UAE black when launched from Arcade?

Does this version improve anything? http://fs-uae.net/devel/2.7.7dev2/

**Comment by MatteAce** [comment]
Arcade works correctly and I can browse all the games. When I launch a game
the screen goes completely black, no ALT+TAB, WIN+D, etc. but I can hear
sounds from the game being played. I'm downloading the DEV2 right now and
I'll then report.

2016-02-16 0:12 GMT+01:00 Frode Solheim <notifications@github.com>:

**Comment by MatteAce** [comment]
nope, dev2 still doesn't work.

**Comment by FrodeSolheim** [comment]
OK, thanks for testing (dev2 was primarily built for testing FrodeSolheim/fs-uae-arcade#7, which is another issue).

I have made quite a few more changes to the Arcade for a new version, which may or may not help. I also suspect running FS-UAE Arcade in window/maximized mode instead of fullscreen might help. There'll be an option for that in the next version.

**Comment by MatteAce** [comment]
yeah I was thinking that as well but the older 2.5 is still working correctly for me. I've also tried to look if there was a config file for the arcade but I couldn't find anything anywhere in the Program folder.

While I wait for the new version, may I ask if there's a way to upgrade WHDLOADER on FS-UAE 2.5 so that I can use the unrestricted >18.2 version?

**Comment by FrodeSolheim** [comment]
Not officially, but some clever person (jbl007) did figure out a hack for that: http://eab.abime.net/showpost.php?p=1059471&postcount=4 ;)

**Comment by FrodeSolheim** [comment]
@MatteAce Follow-up question. Is your setup a hybrid GPU-setup, with switchable graphics? (internal on CPU + external GPU) If so, is is possible to temporarily deactivate this (only use either internal or external GPU). If relevant, and possible, does this help?

**Comment by MatteAce** [comment]
it is a hybrid CPU (it's an A8 after all) but I keep it disabled and I have everything hooked up on my R9 380.

**Comment by FrodeSolheim** [comment]
Has FS-UAE 2.7.8.dev improved anything?

**Comment by MatteAce** [comment]
> Has FS-UAE 2.7.8.dev improved anything?

Yes! Sorry I didn't get back to you sooner but it completely slipped my mind!
The arcade is working fine now, except that it always displays the windows taskbar on top of it, and being that I use it through Steam Big Picture it's a bit annoying. 

I've also found a new... bug maybe? but maybe I should report it elsewhere. Basically a bunch of games won't start from the Arcade, leaving the screen with the spinning wheel you get after starting a game while it waits for FS-UAE to load up. Should I open a new ticket?

**Comment by FrodeSolheim** [comment]
Thanks for testing, I really had no idea of 2.7.8dev would improve anything for this issue. To get the fullscreen back, you can set the following in advanced settings in FS-UAE Launcher:

    arcade_fullscreen = 1

The question which remains is... will FS-UAE now become black again, or not?

**Comment by FrodeSolheim** [comment]
For games not starting, ..., opening a ticket is one possibility. In any case, I need to see fs-uae-arcade.log.txt. You can look for errors yourself in this file if you want, or send the log file (after experiencing problems, and quitting the program) to frode@fs-uae.net!

**Comment by mak77** [comment]
2.7.8 wfm, with or without arcade_fullscreen set to 1.

**Comment by MatteAce** [comment]
alright, with fullscreen on I got back straight where we started... with a black screen on a game startup.
games not starting, I've found out myself: I didn't have the A600 kickstart in my folder so it was loading *slowly* from the Launcher and not loading at all from the Arcade!

[Mak!!! mannaggiatte sono Ace!! come si mandano i PM qua? :D]

**Comment by FrodeSolheim** [comment]
@MatteAce I was "afraid" that the fix consisted running FS-UAE Arcade in windowed/maximized mode. It appears that (in some circumstances) opening a fullscreen OpenGL window while another fullscreen OpenGL window is active is buggy with ATI drivers (not confirmed, but I did experience a similar problem with ATI a couple of years ago).

Though a bit strange that is works for @mak77...

**Comment by mak77** [comment]
By looking at the config I noticed I also had a fullscreen_mode = fullscreen... when that is set it works regardless of arcade_fullscreen. Without fullscreen_mode it doesn't work with arcade_fullscreen = 1.

**Comment by FrodeSolheim** [comment]
@mak77 Thanks, that is helpful. I need to consider whether to use fullscreen_mode = fullscreen by default for FS-UAE, when launched from Arcade (on Windows).

**Comment by MatteAce** [comment]
should I try that as well Frode?

**Comment by FrodeSolheim** [comment]
@MatteAce Yes, it would be nice to get a confirmation.

Related issue (consider enabling fullscreen_mode = fullscreen by default) created: https://github.com/FrodeSolheim/fs-uae-arcade/issues/10.

**Comment by MatteAce** [comment]
it doesn't work for me :/
